libcodechecker/server/session_manager.py

```diff
@@ -469,34 +528,49 @@ def get_session(self, token):
 
         for sess in self.__sessions:
             if sess.is_alive and sess.token == token:
+                # If the session is alive but the should be re-validated.
+                if sess.is_refresh_time_expire:
+                    sess.revalidate()
                 return sess
 
         # Try to get a local session from the database.
-        local_session = self._get_local_session_from_db(token)
+        local_session = self.__get_local_session_from_db(token)
         if local_session and local_session.is_alive:
             self.__sessions.append(local_session)
-        else:
-            self.invalidate(token)
+            if local_session.is_refresh_time_expire:
+                local_session.revalidate()
+            return local_session
+
+        self.invalidate(token)
+        return None
 
         return local_session
 
+    def invalidate_local_session(self, token):
+        """
+        Remove a user's previous local session from the store.
+        """
+        for session in self.__sessions[:]:
+            if session.token == token:
+                self.__sessions.remove(session)
+                return True
+        return False
+
     def invalidate(self, token):
         """
         Remove a user's previous session from the store.
         """
         try:
+            self.invalidate_local_session(token)
```

Comment:
Please express in more detail that `invalidate` will remove the local token and the token from the database but `invalidate_local_session` will just remove the token from the local in memory store.